src/main/java/edu/byu/cs/autograder/compile/ModuleIndependenceVerifier.java

```diff
@@ -48,18 +53,9 @@ private void checkImports(GradingContext context, StudentCodeReader reader, Set<
             for (int i = 0; i < contents.size(); i++) {
                 String line = contents.get(i).trim();
                 if (line.isEmpty()) continue;
-                if (!(line.startsWith("package") || line.startsWith("import"))) {
-                    break;
-                }
-
-                String packageImport = line.substring(line.indexOf(' ')).trim();
-                if (packageImport.startsWith("static")) {
-                    packageImport = packageImport.substring(packageImport.indexOf(' '), packageImport.lastIndexOf('.')).trim();
-                }
-                if (packageImport.contains(".")) {
-                    packageImport = packageImport.substring(0, packageImport.lastIndexOf('.'));
-                }
+                if (line.matches("^(/\\*\\*|@|(?!import|package|//).*\\b(class|record|enum|@?interface)\\b).*$")) break;
```

Comment:
1. These seem equally clear to me, but if putting it the negative lookahead first would be clearer to you I'm amenable to that.
2. You would think that, but empirically it does actually matter. Either 1-Regex101 is incorrect, 2-Something changed in java between Java 8 (Regex101) and Java 21 (autograder) that matters here, or 3-something is different about my implementation and the test cases on Regex101

Reply:
I now understand that in Java, the regex matcher is looking to match the entire line. Since that's the case, it is necessary to include the `.*$` which also clarifies the role of the entire regex.

Reply:
In a discussion with @19mdavenport, we decided to migrate away from this "shortcut" approach because it has the possibility of many edge cases that could misbehave. Instead, we'll scan the whole file, but only pay attention to lines that, when trimmed, begin with "import" and then attempt to extract the import statement.